src/hotspot/os/linux/cgroupV1Subsystem_linux.cpp

```diff
@@ -72,12 +73,40 @@ void CgroupV1Controller::set_subsystem_path(char *cgroup_path) {
             buf[MAXPATHLEN-1] = '\0';
             _path = os::strdup(buf);
           }
+        } else {
+          // Find the longest common prefix as namespaces are hierarchical.
+          // If there are no matches between the root and the cgroup path,
+          // the controller's mount point will be used, which is a reasonable
+          // fallback.
+          stringStream ss;
+          ss.print_raw(_mount_point);
+          const char* root_p = _root;
+          const char* cgroup_p = cgroup_path;
+          int last_slash = find_last_slash_pos(root_p, cgroup_p);
+          assert(last_slash >= 0, "not an absolute path?");
```

Comment:
Are root_p and cgroup_p directly read from the /proc/xxx files. If so, do we validate the input to make sure they are absolute paths?

It seems like our code cannot handle trailing '/' in the input. If so, we should clear all trailing '/' from the input string. Then, in functions that process them, we should assert that they don't end with slash. See my comment in find_last_slash_pos().

Reply:
Yes, those values come from `/proc/self/mountinfo` and `/proc/self/cgroup`. There is no validation being done. Then again, we only end up in this branch if the root path is not a substring of the cgroup path. In that case trailing slashes don't matter, since there would not be a character by character match earlier.

I'll add handling of trailing slashes and appropriate asserts where it makes sense.